index.bs

```diff
@@ -4894,16 +4896,16 @@ function process(numberOfFrames) {
 
 	// Render each sample frame in the quantum
 	for (let index = 0; index &lt; numberOfFrames; index++) {
-		// Check that currentTime is within allowable range for playback
-		if (currentTime &lt; start || currentTime >= stop) {
+		// Check that currentTime and bufferDuration are within allowable range for playback
+		if (currentTime &lt; start || currentTime >= stop || bufferDuration >= duration) {
```

Comment:
I think this is wrong?  Let duration = 129/sampleRate and let the playbackRate = 1.  Since bufferDuration is initialized to 0 in process(), and framesToProcess is 128, bufferDuration is at most 128/sampleRate.  So bufferDuration >= duration is never true.  This means the following lines produce an output.  I think that's wrong.  Maybe this should be bufferTime instead?

Reply:
Perhaps `bufferDuration` was intended to be declared outside the function?

Reply:
Thanks @karlt for catching my error here. Now fixed.